cmd/kubeadm/app/cmd/certificatekey.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/lithammer/dedent"
+	"github.com/spf13/cobra"
+	"k8s.io/kubernetes/cmd/kubeadm/app/phases/copycerts"
+	kubeadmutil "k8s.io/kubernetes/cmd/kubeadm/app/util"
+)
+
+// NewCmdCertificateKey returns cobra.Command for certificate key generate
+func NewCmdCertificateKey() *cobra.Command {
+	return &cobra.Command{
+		Use:   "certificate-key",
```

Comment:
hm, i wonder if we should just use the init phase for this.
adding a top level command would be fine if has more than one sub commands and multiple flags, but right now it would only serve this single purpose.

on the other hand we already have `kubeadm token` which covers existing functionality in `init`.

we also have the option here to have `kubeadm certificate-key` do multiple things:
- only generate key
- upload the certs & generate key (duplicates what the init phase does)

Reply:
also, a problem related to UX that i've been seeing is that users want the new full join command when generating a new cert-key.

so they use `token ... --print-join-command` and combine it with the new cert key manually.

there are no good solutions for this unless we have a command that generates both a new bootstrap token + hash + cert key.

Reply:
this command just generate a certificate-key, so user can only use it on init and init print the join command, we can print the init command but `token generate` just print the token.

Reply:
due that the certificate-key can't be manually generate on a easy way, it would be nice to have a command for it, but I agree that a top level command only for this seems weird.

Reply:
I agree with @neolit123 . My idea is that this is an utility for certificates and should go with the rest of the certs commands. Probably adding it under `kubeadm alpha certs` is a good idea.
@fabriziopandini @yagonobre @neolit123 WDYT?

Reply:
+1 to move to `kubeadm alpha certs`

Reply:
+1